integr8ly/application-monitoring-operator
# INTLY-2597/INTLY-2351 - retention & storage for metrics data

https://issues.jboss.org/browse/INTLY-2597

2 new fields are added to the ApplicationMonitoring CRD

* prometheusRetention
* prometheusStorageRequest

These allow the retention time & storage claim size to be (optionally) set.

The use case for integreatly is to set a retention of 45d, and a storage claim size depending on the expected size of data in OSD clusters (TBD).
However, the integreatly values shouldn't influence the defaults.
By default, there should be no storage claim. The retention default can be set to the prometheus default of 15d.


To verify:

* follow the instructions to run the operator locally against a remote cluster using `make code/run`
* verify the prometheus entrypoint inclucdes an arg `--storage.tsdb.retention=15d`
* verify the prometheus pod is mounting a PVC of size 10Gi
* verify metrics are retained after a restart of promtheus (can delete the prometheus pod and wait for a new one to start)
